libdnf/sack/query.cpp

```diff
@@ -63,20 +63,51 @@ struct default_delete<DnfPackage> {
 namespace libdnf {
 
 struct NevraID {
-    Id name;
-    Id arch;
-    Id evr;
+public:
+    NevraID();
+    NevraID(const NevraID & src);
+    NevraID(NevraID && src);
+    NevraID & operator=(const NevraID & src);
+    NevraID & operator=(NevraID && src);
+    Id name{0};
+    Id arch{0};
+    Id evr{0};
+    std::string evr_str;
     /**
     * @brief Parsing function for nevra string into name, evr, arch and transforming it into libsolv
     * Id
     *
+    * int createNewEVR - `1` will create new id for evr when it is unknown, `0` will exit with false when evr is unknown
+    *
     * @return bool Returns true if parsing succesful and all elements is known to pool
     */
-    bool parse(Pool * pool, const char * nevraPattern);
+
+    bool parse(Pool * pool, const char * nevraPattern, bool createEVRId);
 };
 
+NevraID::NevraID() = default;
+
+NevraID::NevraID(const NevraID & src) = default;
+
+NevraID::NevraID(NevraID && src)
+: name(src.name), arch(src.arch), evr(src.evr), evr_str(std::move(src.evr_str))
+{}
+
+NevraID &
+NevraID::operator=(const NevraID & src) = default;
+
+NevraID &
+NevraID::operator=(NevraID && src)
```

Comment:
Why you define your own move constructor? Is it better than using default one (you can explicitly write `=defaut`)

Reply:
Thanks, removed.